core/src/main/java/org/apache/accumulo/core/data/Mutation.java

```diff
@@ -318,8 +318,8 @@ private void put(byte[] cf, int cfLength, byte[] cq, int cqLength, byte[] cv, bo
     if (buffer == null) {
       throw new IllegalStateException("Can not add to mutation after serializing it");
     }
-    estimatedSize += cfLength + cqLength + (hasts ? 8 : 0) + valLength + 2 * 1 + 4 * SERIALIZATION_OVERHEAD;
-    Preconditions.checkArgument(estimatedSize < MAX_MUTATION_SIZE && estimatedSize >= 0, "Maximum mutation size must be less than 2GB ");
+    long estimatedSizeAfterPut = estRowAndLargeValSize + buffer.size() + cfLength + cqLength + (hasts ? 8 : 0) + valLength + 2 * 1 + 4 * SERIALIZATION_OVERHEAD;
```

Comment:
Should include `cv.length` in the sum.  

Is `2*1` for the two booleans (deleted and hasts)?   This could be simplified to `2`.